enhancements/hypershift/priority-class/priority-class.md

```diff
@@ -0,0 +1,266 @@
+    description: "Used on master ETCD resources for cruisers with 51-100 workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-xl
+    value: 5000000
+    globalDefault: false
+    description: "Used on master control plane resources for cruisers with 101-300 workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-etcd-xl
+    value: 5500000
+    globalDefault: false
+    description: "Used on master ETCD resources for cruisers with 101-300 workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-xxl
+    value: 6000000
+    globalDefault: false
+    description: "Used on master control plane resources for cruisers with 300+ workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-etcd-xxl
+    value: 6500000
+    globalDefault: false
+    description: "Used on master ETCD resources for cruisers with 300+ workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-dedicated
+    value: 7000000
+    globalDefault: false
+    description: "Used on master control plane resources for dedicated cruisers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-etcd-dedicated
+    value: 7500000
+    globalDefault: false
+    description: "Used on master ETCD resources for dedicated cruisers."
+```
+
+With the availablity of the above priority classes, HO can be enhanced to use and override priority classes being on set on the underlying control plane workloads/pods.
+
+### Workflow Description
+TBD
+### API Extensions
+HC and HCP CRDs -- new field
+
+### Implementation Details/Notes/Constraints
+This effort will be implemented over two phases:
+- Phase 1 - Override Priority Classes
+- Phase 2 - Dynamic Adjustment of Priority Classes
+
+### Phase 1 - Override Priority Classes
+Hypershift currently creates and uses a static set of default priority classes. The first phase of this effort will largely focus on Hypershift's ability to override these priority classes for underlying workloads.
+- The overridden priority classes must be specified in the newly created API fields for the HC. 
+- The HC will have three new API fields for the three various priority classes used by hypershift currently--ControlPlanePriorityClass, APICriticalPriorityClass, and EtcdPriorityClass.
+- Override criteria:
+  - Adding a `ControlPlanePriorityClass` in the HC spec will propogate to/override any components with the `hypershift-control-plane` priority class.
+  - Adding a `APICriticalPriorityClass` in the HC spec will propogate to/override any components with the `hypershift-api-critical` priority class.
+  - Adding a `EtcdPriorityClass` in the HC spec will propogate to/override any components with the `hypershift-etcd` priority class.
```

Comment:
nit: Pods have [`priorityClassName`][1].  Could you just add that same property under [`EtcdSpec`][2]?  It seems like the distinction between "control plane" and "api-critical" is explained [here][3], but I don't see any existing structure under `HostedClusterSpec` that seems to break down that way.  Possibly it's worth growing that structure?  Because there may be other `PodSpec`-like things that need configuring for those separate workloads?

[1]: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#pod-priority
[2]: https://github.com/openshift/hypershift/blob/d1d6eee44555145f0b805173a37cda5c5c6f9836/api/v1beta1/hostedcluster_types.go#L1322-L1342
[3]: https://github.com/openshift/hypershift/blob/122678ba9a81b147474bfad00479f263376b0671/support/config/constants.go#L7-L14

Reply:
We've changed the approach around to use HostedCluster annotations instead of our initial approach with API additions. Will leave this marked unresolved as a reminder if we switch the approach back to using the `HostedClusterSpec`.